src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1544,6 +1551,14 @@ trait Namers extends MethodSynthesis {
             if (!isConstr)
               methOwner.resetFlag(INTERFACE) // there's a concrete member now
             val default = parentNamer.enterSyntheticSym(defaultTree)
+            if (meth.name == nme.apply && meth.hasFlag(CASE)) {
```

Comment:
Technically, right now we could restrict it to methods that also have the `SYNTHETIC` flag, right?